src/ChatWork/Utils.hs

```diff
@@ -13,45 +15,127 @@ module ChatWork.Utils (
     , strLength
     ) where
 
+import           Control.Exception            (bracket, handle, mask_, throwIO,
+                                               try)
+import           Control.Monad                ((>=>))
 import           Control.Monad.IO.Class       (MonadIO (..))
+import           Control.Retry                (retrying)
+import           Data.ByteString              (ByteString)
+import qualified Data.ByteString              as B
 import           Data.Default.Class           (def)
+import           Data.IORef                   (modifyIORef', newIORef,
+                                               readIORef, writeIORef)
 import           Data.List                    (lookup)
 import           Data.Maybe                   (fromMaybe)
 import           Data.Proxy                   (Proxy (..))
 import           Network.Connection           (initConnectionContext)
 import           Network.HTTP.Client          (BodyReader, Manager,
                                                ManagerSettings (..), Request,
-                                               Response (..), newManager)
-import           Network.HTTP.Client.Internal (constBodyReader)
-import           Network.HTTP.Client.TLS      (mkManagerSettingsContext)
+                                               Response (..))
+import qualified Network.HTTP.Client          as L
+import qualified Network.HTTP.Client.Internal as LI
+import qualified Network.HTTP.Client.TLS      as L
 import           Network.HTTP.Req             (AllowsBody (..),
                                                CanHaveBody (..),
+                                               HttpConfig (..),
+                                               HttpException (..),
                                                HttpMethod (..),
-                                               HttpResponse (..), MonadHttp)
+                                               HttpResponse (..),
+                                               MonadHttp (..), getHttpConfig)
 import           Network.HTTP.Types           (methodDelete)
 import           Network.HTTP.Types.Header    (hContentLength)
 
 -- |
 -- Helper function that use custamized Manager
 getHttpResponse' :: (HttpResponse a, MonadHttp m) => Proxy a -> Request -> Manager -> m a
-getHttpResponse' Proxy r _ = liftIO $ getHttpResponse r =<< fixEmptyStringManager
+getHttpResponse' Proxy request _ = do

```

Comment:
Maybe I'm missing something, but you could use an alternative manager via `httpConfigAltManager` setting in `HttpConfig`. It's a shame you have to copy so much from Req to get things done...

Reply:
Thank you for advising me.
I will fix that code later.